src/mgr/Action.h

```diff
@@ -76,11 +78,16 @@ class Action: public RefCountable
     /// calculate and keep local action-specific information
     virtual void collect() {}
 
+    /// write manager report output to a stream.
+    virtual void print(std::ostream &) {}
+
     /** start writing action-specific info to Store entry;
      * may collect info during dump, especially if collect() did nothing
      * non-atomic() actions may continue writing asynchronously after returning
+     *
+     * \deprecated implement collect() and print() instead
```

Comment:
Kind of do want to discuss refactoring designs for some of them. But later when they are converted.
This line currently is for new reports being created - to help separate the async actions as `collect()` and display as atomic(ish) `print()`.

Reply:
> This line currently is for new reports being created

I do not agree with the proposed advice for new reports (because not all reports can separate collection of information from printing it).

I am happy to discuss "refactoring designs" later, but that agreement to discuss something later does not address this change request about the current PR comment.

Reply:
This change request was resolved but not addressed. The suggested replacement is now `implement report() instead`, but the essence of this change request has not changed.

Reply:
Oops, forgot to push after committing that change. done in 30e31700df9ca6abcab24a8c768b9b95a81072f2